### PR TITLE
Document entity inheritance and add integration test

### PR DESCRIPTION
**Pull request by mqus**
This should address #342 and prevent similar issues.

### CODE REVIEW

File: floor/test/integration/inheritance/entity_inheritance_test.dart [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    setUp(() async {
+      database = await $FloorTestDatabase.inMemoryDatabaseBuilder().build();
+      commentDao = database.commentDao;
+    });
+
+    tearDown(() async {
+      await database.close();
+    });
+
+    test('use generated inherited insert function', () async {
```
Comment (vitusortner) [inline floor/test/integration/inheritance/entity_inheritance_test.dart:24]: ⚠️ The test's name doesn't match the test

File: floor/test/integration/inheritance/entity_inheritance_test.dart [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+import 'package:floor/floor.dart';
+import 'package:floor_annotation/floor_annotation.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:sqflite/sqflite.dart' as sqflite;
+
+part 'entity_inheritance_test.g.dart';
+
+void main() {
+  group('dao inheritance tests', () {
```
Comment (vitusortner) [inline floor/test/integration/inheritance/entity_inheritance_test.dart:11]: ⚠️ The name doesn't match the test group

File: floor/test/integration/inheritance/entity_inheritance_test.dart (hunk 1 above)
Comment (mqus) [reply floor/test/integration/inheritance/entity_inheritance_test.dart]: whoops

File: floor/test/integration/inheritance/entity_inheritance_test.dart (hunk 2 above)
Comment (mqus) [reply floor/test/integration/inheritance/entity_inheritance_test.dart]: whoops

File: README.md [hunk 3]
```diff
@@ -319,6 +319,49 @@ class Person {
   Person(this.id, this.name);
 }
 ```
+
+### Inheritance
```
[the remark this answers is not in the document]
Comment (mqus) [reply README.md:324]: sure!

**Pull request merged**